tests/rpc/rpc_test.go

```diff
@@ -644,8 +646,15 @@ func TestEth_GetTransactionReceipt_ContractDeployment(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "0x1", receipt["status"].(string))
 
-	require.NotEqual(t, common.Address{}.String(), receipt["contractAddress"].(string))
-	require.NotNil(t, receipt["logs"])
+	addr := common.BytesToAddress(common.Hex2Bytes(receipt["contractAddress"].(string)))
+	require.Equal(t, crypto.CreateAddress(common.BytesToAddress(from), uint64(nonce)), addr)
+	require.Greater(t, len(receipt["logs"].([]interface{})), 0)
+
+	rpcRes = call(t, "eth_getCode", []string{addr.Hex(), zeroString})
```

Comment:
Can you add a test on the tests/integration/e2e.go file instead so that we use the rpc client directly?

Reply:
sure, hold on.

Reply:
how to sign a tx in e2e integration test, do I need to hard code the private key?

Reply:
see the comment for the file to add the integration test into

Reply:
you can generate one using `tests.NewAddrKey()`

Reply:
> > how to sign a tx in e2e integration test, do I need to hard code the private key?
> 
> you can generate one using `tests.NewAddrKey()`

But then I need to send test tokens to it, I still need a key that have preallocated tokens.

> see the comment for the file to add the integration test into

is it [this file](https://github.com/tharsis/ethermint/blob/main/tests/e2e/integration_test.go)? I don't see much comments there.

Reply:
yeah, it uses the JSON-RPC `Client` directly instead of doing an RPC call. It only has a few tests but we plan to migrate the `test-rpc` ones to that file.

Reply:
it seems to me that the roc test is more superior because it don't go through IPC and probably report produce a better coverage report?
How is the test coverage work when the node is running as a standalone process?

Reply:
When I try to port the test, there's no `eth_sendTransaction` API in the RPC client, there's only `eth_sendRawTransaction`, with which we need to sign with a hardcoded private key.

Reply:
`s.network.Validators[0].JSONRPCClient.SendTransaction(ctx, tx)` should work, no?

Reply:
> `s.network.Validators[0].JSONRPCClient.SendTransaction(ctx, tx)` should work, no?

No, that one calls "sendRawTransaction", so we need to sign the tx locally.